### ISSUE TITLE
[BUG]: Error while updating 'SelectedTextTrack' 

### ISSUE THREAD

**Issue by rahulnainwalttn** [TheWidlarzGroup/react-native-video#3704]
### Version

v6 (Beta)

### What platforms are you having the problem on?

Android

### Architecture

Old architecture

### What happened?

My text tracks looks like below JSON. Initially, index 1 is selected. when I am selecting 2 or 3 indexes its working fine but as soon i am selecting index 0 app is getting crashed in **[6.0.0-rc.0](https://github.com/react-native-video/react-native-video/compare/v6.0.0-beta.8...v6.0.0-rc.0)**

Stream URL - https://bitmovin-a.akamaihd.net/content/sintel/hls/playlist.m3u8

"textTracks": [{"index": 0, "language": "de", "selected": false, "title": "Deutsch", "type": "text/vtt"}, {"index": 1, "language": "en", "selected": true, "title": "English", "type": "text/vtt"}, {"index": 2, "language": "es", "selected": false, "title": "Espanol", "type": "text/vtt"}, {"index": 3, "language": "fr", "selected": false, "title": "Français", "type": "text/vtt"}]

![Image_20240423_171324_526](https://github.com/react-native-video/react-native-video/assets/136161024/ae9f6c66-9168-4ebc-96a2-7e617f2036fd)

### Reproduction

repository link

### Reproduction

Step to reproduce this bug are:

**Comment by rahulnainwalttn** [comment]
Logs - 

1 - 
```
Error while updating prop selectedTextTrack
                                                                                                    java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:93)
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136)
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56)
                                                                                                    	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48)
                                                                                                    	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086)
                                                                                                    	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
                                                                                                    	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
                                                                                                    	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
                                                                                                    Caused by: java.lang.NumberFormatException: s == null
                                                                                                    	at java.lang.Integer.parseInt(Integer.java:759)
                                                                                                    	at java.lang.Integer.parseInt(Integer.java:915)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.setSelectedTrack(ReactExoplayerView.java:1665)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.setSelectedTextTrack(ReactExoplayerView.java:1833)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerViewManager.setSelectedTextTrack(ReactExoplayerViewManager.java:284)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:93) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56) 
                                                                                                    	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48) 
                                                                                                    	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086) 
                                                                                                    	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29) 
                                                                                                    	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175) 
                                                                                                    	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659) 
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129) 
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045) 
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                                                    	at android.os.Looper.loop(Looper.java:319) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 
2024-04-23 17:03:47.530 22890-22890 
```

2- 
```
E  Exception in native call
                                                                                                    com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'selectedTextTrack' of a view managed by: RCTVideo
                                                                                                    	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:101)
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136)
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56)
                                                                                                    	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48)
                                                                                                    	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086)
                                                                                                    	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
                                                                                                    	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
                                                                                                    	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648)
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659)
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129)
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045)
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230)
                                                                                                    	at android.os.Looper.loop(Looper.java:319)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
                                                                                                    Caused by: java.lang.reflect.InvocationTargetException
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:93)
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56) 
                                                                                                    	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48) 
                                                                                                    	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086) 
                                                                                                    	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29) 
                                                                                                    	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175) 
                                                                                                    	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659) 
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129) 
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045) 
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                                                    	at android.os.Looper.loop(Looper.java:319) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 
                                                                                                    Caused by: java.lang.NumberFormatException: s == null
                                                                                                    	at java.lang.Integer.parseInt(Integer.java:759)
                                                                                                    	at java.lang.Integer.parseInt(Integer.java:915)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.setSelectedTrack(ReactExoplayerView.java:1665)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerView.setSelectedTextTrack(ReactExoplayerView.java:1833)
                                                                                                    	at com.brentvatne.exoplayer.ReactExoplayerViewManager.setSelectedTextTrack(ReactExoplayerViewManager.java:284)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:93) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136) 
                                                                                                    	at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56) 
                                                                                                    	at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:48) 
                                                                                                    	at com.facebook.react.uimanager.NativeViewHierarchyManager.updateProperties(NativeViewHierarchyManager.java:142) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$UpdatePropertiesOperation.execute(UIViewOperationQueue.java:93) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:915) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47) 
                                                                                                    	at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086) 
                                                                                                    	at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29) 
                                                                                                    	at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175) 
                                                                                                    	at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1648) 
                                                                                                    	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1659) 
                                                                                                    	at android.view.Choreographer.doCallbacks(Choreographer.java:1129) 
                                                                                                    	at android.view.Choreographer.doFrame(Choreographer.java:1045) 
                                                                                                    	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1622) 
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:958) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:230) 
                                                                                                    	at android.os.Looper.loop(Looper.java:319) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8893) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103) 
```

**Comment by rahulnainwalttn** [comment]
Same is happening with audio update - 
![Image_20240423_174930_411](https://github.com/react-native-video/react-native-video/assets/136161024/b6249462-2cf9-4219-98b7-3a8e0c32435f)

**Comment by mkbhdana** [comment]
it arises while setting index as selectedTrack. You can convert index to string it will solve the problem

**Comment by rahulnainwalttn** [comment]
Is it not valid anymore with new version?
<br/>
<img width="849" alt="Screenshot 2024-04-23 at 6 56 19 PM" src="https://github.com/react-native-video/react-native-video/assets/136161024/a404e48b-19de-43cb-977a-6b2a6f208a42">

**Comment by KrzysztofMoch** [comment]
No, it should work - I will fix this

**Comment by rahulnainwalttn** [comment]
Thanks @KrzysztofMoch

**Comment by mkbhdana** [comment]
@KrzysztofMoch is there anyway that we can use exoplayer ffmpeg extension in react native video as some formats are not playable

can i share you file url which you can try as the file is hevc still it does not play

**Comment by freeboub** [comment]
> @KrzysztofMoch is there anyway that we can use exoplayer ffmpeg extension in react native video as some formats are not playable
> 
> can i share you file url which you can try as the file is hevc still it does not play

No, currently not, I tried to enable it, but exoplayer doesn't support it as a plug in anymore.
To achieve that you need to rebuild exoplayer... (Not 100% sure with media3 I checked with exoplayer 2.18 If I remember well)
https://github.com/react-native-video/react-native-video/issues/3106

**Comment by freeboub** [comment]
I just open a PR to fix this issue. If you want to test it it would be great !
https://github.com/react-native-video/react-native-video/pull/3721
Thank youi

**Comment by rahulnainwalttn** [comment]
Thanks @freeboub. Sure will test it and confirm.

**Comment by rahulnainwalttn** [comment]
Hi @freeboub Android issue is fixed but in ios switching between audio tracks is not working and subtitles are not visible in ios (I'm not sure if it is working or not). 

Tested using the same stream URL.

**Comment by rahulnainwalttn** [comment]
@freeboub Any update on it?

**Comment by freeboub** [comment]
@rahulnainwalttn I was working on the issue, I have just push a pull request !

**Comment by rahulnainwalttn** [comment]
Perfect @freeboub. It's working now. Thanks

**Comment by freeboub** [comment]
Let's close this ticket, Thank you for the feedback.
Fix will be available ion 6.0.0-RC.1
